src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs

```diff
@@ -209,7 +209,11 @@ public async Task<int> ExecuteCommand(PackageReferenceArgs packageReferenceArgs,
                     packageReferenceArgs.PackageId,
                     packageReferenceArgs.ProjectPath));
 
-                var compatibleOriginalFrameworks = originalPackageSpec.RestoreMetadata
+                var compatibleOriginalFrameworks = compatibleFrameworks
+                    .Select(e => GetMatchingFrameworkStrings(originalPackageSpec, e))
+                    .Where(originalFramework => originalFramework != null);
+
+                    originalPackageSpec.RestoreMetadata
```

Comment:
I don't understand what is going on here? You filtered something but not assigned to anything, so it'll be lost.

Reply:
It's used in a line that I did not change, so it won't show up in the diff. 
See https://github.com/NuGet/NuGet.Client/blob/e1fb834826d4ee8203dcefda767df107eb25e716/src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs#L219-L221

Reply:
I think @erdembayar's right here. Line 214 ends in a semicolon, and this line starts a new LINQ query, but the result is not used anywhere, and therefore will never be evaluated.

Reply:
Yes, I think you need to replace replace `;` with `.` in Line 214.

Reply:
It took me a while to get what the feedback was. 

No need for `.`. 
The code I added replaces that. That's dead code. 

Removed.